crypto/compactcert/weights.go

```diff
@@ -0,0 +1,175 @@
+// Copyright (C) 2019-2022 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package compactcert
+
+import (
+	"errors"
+	"math"
+	"math/big"
+	"math/bits"
+)
+
+// errors for the weights verification
+var (
+	ErrSignedWeightLessThanProvenWeight = errors.New("signed weight is less than or equal to proven weight")
+	ErrTooManyReveals                   = errors.New("too many reveals in cert")
+	ErrZeroSignedWeight                 = errors.New("signed weight can not be zero")
+	ErrIllegalInputForLnApprox          = errors.New("can not calculate a ln integer value for 0")
+	ErrInsufficientSingedWeight         = errors.New("the number of reveals is not large enough to prove that the desired weight signed")
```

Comment:
This might be a little bit misleading since there are multiple signatures in the cert and the "power" of the cert is derived from how much weight is singed (aka the signed weight)
the number of reveals is a "challge" for the claim that this cert has X signed weight.

Reply:
After our discussion, I agree with the original comment. The issue is `weight signed` is not meaningful to someone not familiar with the literature.